core/src/impl/Kokkos_Core.cpp

```diff
@@ -552,88 +582,45 @@ void parse_command_line_arguments(int& narg, char* arg[],
-      std::cout << "the non-prefixed ones are not removed. Prefixed versions "
-                   "take precedence over "
-                << std::endl;
-      std::cout << "non prefixed ones, and the last occurrence of an argument "
-                   "overwrites prior"
-                << std::endl;
-      std::cout << "settings." << std::endl;
-      std::cout << std::endl;
-      std::cout << "--kokkos-help               : print this message"
-                << std::endl;
-      std::cout
-          << "--kokkos-disable-warnings   : disable kokkos warning messages"
-          << std::endl;
-      std::cout
-          << "--kokkos-threads=INT        : specify total number of threads or"
-          << std::endl;
-      std::cout << "                              number of threads per NUMA "
-                   "region if "
-                << std::endl;
-      std::cout << "                              used in conjunction with "
-                   "'--numa' option. "
-                << std::endl;
-      std::cout << "--kokkos-numa=INT           : specify number of NUMA "
-                   "regions used by process."
-                << std::endl;
-      std::cout << "--kokkos-device=INT         : specify device id to be used "
-                   "by Kokkos. "
-                << std::endl;
-      std::cout << "--kokkos-ndevices=INT[,INT] : used when running MPI jobs. "
-                   "Specify number of"
-                << std::endl;
-      std::cout << "                              devices per node to be used. "
-                   "Process to device"
-                << std::endl;
-      std::cout << "                              mapping happens by obtaining "
-                   "the local MPI rank"
-                << std::endl;
-      std::cout << "                              and assigning devices "
-                   "round-robin. The optional"
-                << std::endl;
-      std::cout << "                              second argument allows for "
-                   "an existing device"
-                << std::endl;
-      std::cout << "                              to be ignored. This is most "
-                   "useful on workstations"
-                << std::endl;
-      std::cout << "                              with multiple GPUs of which "
-                   "one is used to drive"
-                << std::endl;
-      std::cout << "                              screen output." << std::endl;
-      std::cout << std::endl;
-      std::cout << "-----------------------------------------------------------"
-                   "---------------------"
-                << std::endl;
-      std::cout << std::endl;
-
-      // Remove the --kokkos-help argument from the list but leave --ndevices
-      if (strcmp(arg[iarg], "--kokkos-help") == 0) {
+    } else if (check_arg(arg[iarg], "--kokkos-help") ||
+               check_arg(arg[iarg], "--help")) {
+      std::string const help_message = R"(
```

Comment:
```suggestion
      auto const help_message = R"(
```
would avoid the intermediate `std::string` to be a bit simpler.